### PR TITLE
{AKS} `az aks update`: don't always use the current kube-proxy config when updating cluster

### PR DESCRIPTION
**Pull request by tyler-lloyd**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

`az aks update`

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 1]
```diff
@@ -627,8 +627,9 @@ def get_kube_proxy_config(self) -> Union[Dict, ContainerServiceNetworkProfileKub
                 )
 
         # try to read the property value corresponding to the parameter from the `mc` object
-        if self.mc and self.mc.network_profile and self.mc.network_profile.kube_proxy_config is not None:
-            kube_proxy_config = self.mc.network_profile.kube_proxy_config
+        if self.decorator_mode == DecoratorMode.CREATE:
```
Comment (paulgmiller) [inline src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:630]: confused by this checking of create won't self.mc.network_profile alsways be empty in create?

**Review by paulgmiller — COMMENTED** [review]
confused abotu create but otherwise seems fine.

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py (hunk 1 above)
Comment (tyler-lloyd) [reply src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:630]: network_profile I do not think is empty on create. for example we using 10.244.0.0/16 as the default pod_cidr from the sdk

**Review by FumingZhang — COMMENTED** [review]
LGTM

Queued a [live test pipeline](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=72801286&view=results) to validate the change.

**Review by FumingZhang — APPROVED** [review]
Test passed.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**